### PR TITLE
Docs generate fixes

### PR DESCRIPTION
**Pull request by kalikiana**
- Build docs and update branch locally
- Mention all needed dependencies
- Show and fail on errors
- Leave (Travis)CI in charge of what to build/ generate
- Fix errors in documentation markup (which were previously hidden)
- Publish schema(s) as well ([poo#56426](https://progress.opensuse.org/issues/56426))

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,9 +17,11 @@ matrix:
       env: SCHEDULER_FULLSTACK=1
     - name: "developer fullstack test"
       env: DEVELOPER_FULLSTACK=1
-    - name: "documentation generation job"
+    - name: "generate and publish documentation"
       if: branch = master AND fork = false
       env: GH_PUBLISH=true
+    - name: "generate documentation (sanity-check)"
+      if: branch = master
```
Comment (okurz) [inline .travis.yml]: why not without any restrictions here, same as for the other tests?

**Comment by Martchus** [comment]
And how do you test this locally?

**Comment by kalikiana** [comment]
> And how do you test this locally?

    ./script/generate-documentation

File: docs/WritingTests.asciidoc [hunk 2]
```diff
@@ -1374,7 +1374,7 @@ describes how downloadable assets can be specified. It is important to note
 that the specified asset is only downloaded once. New versions must be
 supplied as new, unambiguous download target file names.
 
-===== Triggering tests based on an any remote git refspec or open github pull request
+=== Triggering tests based on an any remote git refspec or open github pull request
```
Comment (okurz) [inline docs/WritingTests.asciidoc]: thank you for noticing this.

**Pull request merged**